models/events.go

```diff
@@ -119,6 +121,19 @@ func decodeCadenceEvents(events flow.BlockEvents) (*CadenceEvents, error) {
 		return nil, fmt.Errorf("EVM block can not be nil if transactions are present, Flow block: %d", events.Height)
 	}
 
+	if e.block != nil && len(e.transactions) > 0 {
+		txHashes := evmTypes.TransactionHashes([]gethCommon.Hash{})
```

Comment:
then after here I don't think you need to check transactions length... because you should do that even for empty slice of txs... so it should match empty root

Reply:
Removed the check for transactions length in: https://github.com/onflow/flow-evm-gateway/pull/529/commits/1e18078c9eb9b427e0ee671fd44f30749d9947e6